tools/querytee/proxy_endpoint.go

```diff
@@ -386,3 +408,100 @@ func (r *backendResponse) statusCode() int {
 
 	return r.status
 }
+
+// extractMinTimeFromRequest extracts the minimum time from a Prometheus API request
+// Returns zero time if time parameters cannot be parsed, which causes all backends to handle the query.
+func extractMinTimeFromRequest(r *http.Request) time.Time {
+	if err := r.ParseForm(); err != nil {
+		return time.Time{} // Return zero time on parse error
+	}
+
+	path := r.URL.Path
+
+	switch {
+	case strings.HasSuffix(path, "/api/v1/query_range"):
+		return extractMinTimeFromRangeQuery(r)
+	case strings.HasSuffix(path, "/api/v1/query"):
+		return extractTimeFromInstantQuery(r)
+	default:
+		return extractTimeFromGenericQuery(r) // series, labels, exemplars endpoints all use start/end or time parameters
+	}
+}
+
+// extractMinTimeFromRangeQuery extracts the minimum time from range query parameters
+func extractMinTimeFromRangeQuery(r *http.Request) time.Time {
+	start := r.FormValue("start")
+	end := r.FormValue("end")
+
+	if start == "" && end == "" {
+		return time.Time{} // No time parameters found
+	}
+
+	startTime, _ := parsePrometheusTime(start)
+	endTime, _ := parsePrometheusTime(end)
+
+	// Both times invalid
+	if startTime.IsZero() && endTime.IsZero() {
+		return time.Time{}
+	}
+
+	// Return the earlier time (minimum of start and end)
+	if startTime.IsZero() {
+		return endTime
+	}
+	if endTime.IsZero() {
+		return startTime
+	}
+
+	if startTime.Before(endTime) {
+		return startTime
+	}
+	return endTime
+}
+
+// extractTimeFromInstantQuery extracts time from instant query parameters
+func extractTimeFromInstantQuery(r *http.Request) time.Time {
+	timeParam := r.FormValue("time")
+	if timeParam == "" {
+		return time.Now()
+	}
+
+	t, _ := parsePrometheusTime(timeParam)
+	return t
+}
```

Comment:
Rather than doing this, I would use `promql.FindMinMaxTime` to determine the time range queried, as that method takes into account things like the lookback period and the range on vector selectors.